### ISSUE TITLE
CollectionIndexOnNonIndexedSeq causes ScalaReflectionException

### ISSUE THREAD

**Issue by devkat** [sksamuel/scapegoat#162]
Version: 1.3.1-SNAPSHOT

Exception:
```
Error:scala: Error: class com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq in compiler mirror not found.
scala.ScalaReflectionException: class com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq in compiler mirror not found.
	at scala.reflect.internal.Mirrors$RootsBase.staticClass(Mirrors.scala:123)
	at scala.reflect.internal.Mirrors$RootsBase.staticClass(Mirrors.scala:22)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1$$typecreator1$1.apply(CollectionIndexOnNonIndexedSeq.scala:13)
	at scala.reflect.api.TypeTags$WeakTypeTagImpl.tpe$lzycompute(TypeTags.scala:232)
	at scala.reflect.api.TypeTags$WeakTypeTagImpl.tpe(TypeTags.scala:232)
	at scala.reflect.api.TypeTags$class.typeOf(TypeTags.scala:341)
	at scala.reflect.api.Universe.typeOf(Universe.scala:61)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.isSeq(CollectionIndexOnNonIndexedSeq.scala:13)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:22)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1215)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.internal.Trees$class.traverseMemberDef$1(Trees.scala:1203)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1328)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.api.Trees$Traverser.traverseTrees(Trees.scala:2484)
	at scala.reflect.internal.Trees$class.traverseComponents$1(Trees.scala:1234)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1330)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.internal.Trees$class.traverseComponents$1(Trees.scala:1255)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1330)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.api.Trees$Traverser.traverseTrees(Trees.scala:2484)
	at scala.reflect.internal.Trees$class.traverseComponents$1(Trees.scala:1234)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1330)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1215)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.internal.Trees$class.traverseMemberDef$1(Trees.scala:1203)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1328)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.api.Trees$Traverser$$anonfun$traverseStats$1$$anonfun$apply$1.apply$mcV$sp(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
	at scala.reflect.internal.Trees$class.traverseComponents$1(Trees.scala:1232)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1330)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1208)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.internal.Trees$class.traverseMemberDef$1(Trees.scala:1203)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1327)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at scala.reflect.api.Trees$Traverser$$anonfun$traverseStats$1$$anonfun$apply$1.apply$mcV$sp(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
	at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1326)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.com$sksamuel$scapegoat$InspectionContext$Traverser$$super$traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.continue(Inspection.scala:71)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.continue(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.inspect(CollectionIndexOnNonIndexedSeq.scala:28)
	at com.sksamuel.scapegoat.InspectionContext$Traverser$class.traverse(Inspection.scala:86)
	at com.sksamuel.scapegoat.inspections.collections.CollectionIndexOnNonIndexedSeq$$anon$2$$anon$1.traverse(CollectionIndexOnNonIndexedSeq.scala:9)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer$$anonfun$transform$2$$anonfun$apply$2.apply(plugin.scala:199)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer$$anonfun$transform$2$$anonfun$apply$2.apply(plugin.scala:198)
	at scala.Option.foreach(Option.scala:257)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer$$anonfun$transform$2.apply(plugin.scala:198)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer$$anonfun$transform$2.apply(plugin.scala:196)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer.transform(plugin.scala:196)
	at com.sksamuel.scapegoat.ScapegoatComponent$Transformer.transform(plugin.scala:185)
	at scala.tools.nsc.ast.Trees$Transformer.transformUnit(Trees.scala:147)
	at scala.tools.nsc.transform.Transform$Phase.apply(Transform.scala:30)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$applyPhase$1.apply$mcV$sp(Global.scala:467)
	at scala.tools.nsc.Global$GlobalPhase.withCurrentUnit(Global.scala:458)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:467)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:425)
	at com.sksamuel.scapegoat.ScapegoatComponent$$anon$1.run(plugin.scala:151)
	at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1528)
	at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1513)
	at scala.tools.nsc.Global$Run.compileSources(Global.scala:1508)
	at scala.tools.nsc.Global$Run.compile(Global.scala:1609)
	at xsbt.CachedCompiler0.run(CompilerInterface.scala:115)
	at xsbt.CachedCompiler0.run(CompilerInterface.scala:94)
	at xsbt.CompilerInterface.run(CompilerInterface.scala:22)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:101)
	at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:47)
	at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:41)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1$$anonfun$apply$2$$anonfun$apply$1.apply$mcV$sp(MixedAnalyzingCompiler.scala:51)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1$$anonfun$apply$2$$anonfun$apply$1.apply(MixedAnalyzingCompiler.scala:51)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1$$anonfun$apply$2$$anonfun$apply$1.apply(MixedAnalyzingCompiler.scala:51)
	at sbt.compiler.MixedAnalyzingCompiler.sbt$compiler$MixedAnalyzingCompiler$$timed(MixedAnalyzingCompiler.scala:76)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1$$anonfun$apply$2.apply(MixedAnalyzingCompiler.scala:50)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1$$anonfun$apply$2.apply(MixedAnalyzingCompiler.scala:47)
	at scala.Option.foreach(Option.scala:257)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:47)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:47)
	at scala.Option.foreach(Option.scala:257)
	at sbt.compiler.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:47)
	at sbt.compiler.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:66)
	at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:161)
	at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:161)
	at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:70)
	at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:68)
	at sbt.inc.IncrementalCommon.cycle(IncrementalCommon.scala:33)
	at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:62)
	at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:61)
	at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:92)
	at sbt.inc.Incremental$.compile(Incremental.scala:61)
	at sbt.inc.IncrementalCompile$.apply(Compile.scala:56)
	at sbt.compiler.IC$.compileInternal(IncrementalCompiler.scala:161)
	at sbt.compiler.IC$.incrementalCompile(IncrementalCompiler.scala:132)
	at org.jetbrains.jps.incremental.scala.local.SbtCompiler.compile(SbtCompiler.scala:74)
	at org.jetbrains.jps.incremental.scala.local.LocalServer.compile(LocalServer.scala:26)
	at org.jetbrains.jps.incremental.scala.remote.Main$.make(Main.scala:68)
	at org.jetbrains.jps.incremental.scala.remote.Main$.nailMain(Main.scala:25)
	at org.jetbrains.jps.incremental.scala.remote.Main.nailMain(Main.scala)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.martiansoftware.nailgun.NGSession.run(NGSession.java:319)```

**Comment by JoshRosen** [comment]
It looks like this is occurring in a call to `isSeq` at https://github.com/sksamuel/scapegoat/blob/0f66ecca77cf905503c8bfb3bbbdadc830db1d66/src/main/scala/com/sksamuel/scapegoat/inspections/collections/CollectionIndexOnNonIndexedSeq.scala#L13

This pattern is used elsewhere in Scapegoat. In `ExistsSimplifiableToContains`, I see a similar `isTraversable` helper function at https://github.com/sksamuel/scapegoat/blob/4d3bca641b495f68c7a6adf99ae6eba750a3404b/src/main/scala/com/sksamuel/scapegoat/inspections/collections/ExistsSimplifableToContains.scala#L18.

It looks like you're running Scapegoat in an IDE. Is it possible that you're hitting the same issue as #98? Just wondering if maybe this is an environment / configuration issue on your end vs. a bug in this specific rule since at first glance this rule doesn't look significantly different than existing rules.

**Comment by devkat** [comment]
I'm using scapegoat directly as a compiler plugin (haven't found out to make IntelliJ IDEA run it via the sbt plugin):

```
lazy val commonSettings = Seq(
  …
  scalacOptions ++= Seq(
    …
    "-P:scapegoat:ignoredFiles:.*\\\\.template\\\\.scala", // exclude Play Twirl templates
    "-P:scapegoat:dataDir:" + (crossTarget in Compile).value.getAbsolutePath + "/scapegoat-report"
    …
) ++ addCompilerPlugin("com.sksamuel.scapegoat" %% "scalac-scapegoat-plugin" % "1.3.1-SNAPSHOT")
```

The issue occurs both in IntelliJ IDEA and in sbt on the command line.

A similar wartremover bug (https://github.com/wartremover/wartremover/issues/365) occurred in my codebase (IntelliJ IDEA and command line). I've applied an analogous change to the scapegoat code, this seems to fix the issue for me. I didn't check whether the CollectionIndexOnNonIndexedSeq inspection still works, but the tests succeed.

I will submit a PR.

**Comment by devkat** [comment]
I added a test statement to trigger `ExistsSimplifiableToContains`, but it doesn't result in an error:

```
    def test = (1 to 5).exists(_ == 3)
```

Maybe for some reason `ExistsSimplifiableToContains` is not invoked, will have to dig a bit deeper.

**Comment by JoshRosen** [comment]
I believe that this issue can be closed now that the fix in #163 has been merged?
